kernel/spinlock_validate.c

```diff
@@ -20,10 +20,17 @@ bool z_spin_lock_valid(struct k_spinlock *l)
 
 bool z_spin_unlock_valid(struct k_spinlock *l)
 {
-	if (l->thread_cpu != (_current_cpu->id | (uintptr_t)_current)) {
+	uintptr_t tcpu = l->thread_cpu;
+
+	l->thread_cpu = 0;
+
+	if (arch_is_in_isr() && _current->base.thread_state & _THREAD_DUMMY) {
+		/* Edge case where an ISR aborted _current */
+		return true;
+	}
+	if (tcpu != (_current_cpu->id | (uintptr_t)_current)) {
```

Comment:
Will this will limit the core number up to 8 for 64 bit machine, 4 cores for 32 bit machine?

Reply:
Yes, that's an existing limitation of CONFIG_SPIN_VALIDATE.  It would be easy enough to fix, but the cost would be that enabling spinlock checking would change the size of the struct, which gets packed into all sorts of things.  IMHO that's a bad trade for debug/test features.